WebIDL Constructors: add tests to constructors with required arguments which, when called as functions, throw an error

JakeChampion opened the pull request:
This cannot be tested generically (see #34157 for work on that) as the first argument is required.

JakeChampion committed:
FormDataEvent: add test to confirm calling FormDataEvent constructor without new throws an error

This cannot be tested generically (see #34157 for work on that) as the first argument is required.

JakeChampion committed:
ImageData: add test to confirm calling ImageData constructor without new throws an error

This cannot be tested generically (see #34157 for work on that) as the first argument is required.

JakeChampion committed:
OffscreenCanvas: add test to confirm calling OffscreenCanvas constructor without new throws an error

This cannot be tested generically (see #34157 for work on that) as the first argument is required.

JakeChampion committed:
PopStateEvent: add test to confirm calling PopStateEvent constructor without new throws an error

This cannot be tested generically (see #34157 for work on that) as the first argument is required.

JakeChampion committed:
ErrorEvent: add test to confirm calling ErrorEvent constructor without new throws an error

This cannot be tested generically (see #34157 for work on that) as the first argument is required.

JakeChampion committed:
BroadcastChannel: add test to confirm calling BroadcastChannel constructor without new throws an error

This cannot be tested generically (see #34157 for work on that) as the first argument is required.

JakeChampion commented:
@domenic and @annevk , I am doing this for all the constructors which have required arguments - should I do these as separate pull-requests?

domenic commented:
I think a single PR would be easier to review, personally!

However, I'm not sure if we need that extra coverage for every required-arg constructor. We can probably assume that if a browser implements no-required-arg constructors correctly in this regard, they'll also get required-arg constructors correct.

In cases like this, where there are already tests for the constructor, just adding one more line to the file is reasonable. But I'm not sure we need to do it everywhere.

JakeChampion commented:
> I think a single PR would be easier to review, personally!
> 
> However, I'm not sure if we need that extra coverage for every required-arg constructor. We can probably assume that if a browser implements no-required-arg constructors correctly in this regard, they'll also get required-arg constructors correct.
> 
> In cases like this, where there are already tests for the constructor, just adding one more line to the file is reasonable. But I'm not sure we need to do it everywhere.

Would it be okay if I push the commits I already have done for a bunch of the required arg constructors to this branch/pull-request?

JakeChampion committed:
StorageEvent: add test to confirm calling StorageEvent constructor without new throws an error

This cannot be tested generically (see #34157 for work on that) as the first argument is required.

domenic commented:
Sure, no problem.

JakeChampion committed:
PromiseRejectionEvent: add test to confirm calling PromiseRejectionEvent constructor without new throws an error

This cannot be tested generically (see #34157 for work on that) as the first argument is required.

JakeChampion committed:
EventSource: add test to confirm calling EventSource constructor without new throws an error

This cannot be tested generically (see #34157 for work on that) as the first argument is required.

JakeChampion committed:
MessageEvent: add test to confirm calling MessageEvent constructor without new throws an error

This cannot be tested generically (see #34157 for work on that) as the first argument is required.

foolip commented on eventsource/dedicated-worker/eventsource-constructor-no-new.js:
```diff
@@ -0,0 +1,7 @@
+test(function() {
```
This file won't be recognized a test, you'll need to name it .any.js or .window.js

JakeChampion replied:
Thanks @foolip

JakeChampion committed:
remove debugger statements

hoch reviewed (approved):
webaudio/resources/audit.js LGTM

JakeChampion commented:
@domenic I've rebased this and believe it is ready for another round of reviews 👍

domenic commented on html/canvas/element/drawing-text-to-the-canvas/2d.text.draw.fill.maxWidth.large.png:
I don't know why this file was changed.

JakeChampion replied:
Me neither - I've restored the original 👍

domenic commented on css/css-ui/webkit-appearance-auto-non-html-namespace-001.html:
This file seems likely added by mistake?

JakeChampion replied:
It appears if I run `./css/css-ui/tools/appearance-build-webkit-reftests.py`
I've removed the file and rebased 👍

inexorabletash commented on webstorage/event_constructor.window.js:
```diff
@@ -1,3 +1,11 @@
+        TypeError,
+        () => StorageEvent(""),
+        "Calling StorageEvent constructor without 'new' must throw"
+    );
```
Add test description

inexorabletash commented on html/webappapis/scripting/events/messageevent-constructor.https.html:
```diff
@@ -4,6 +4,14 @@
+        () => MessageEvent(""),
+        "Calling MessageEvent constructor without 'new' must throw"
+    );
+});
```
Add test description

inexorabletash commented on html/webappapis/scripting/processing-model-2/unhandled-promise-rejections/promise-rejection-event-constructor.html:
```diff
@@ -9,6 +9,12 @@
   var p = new Promise(function(resolve, reject) {});
 
+  assert_throws_js(TypeError,
+  function() {
```
Indent the arguments to `assert_throws_js` as is done later in the file.

inexorabletash commented on webmessaging/broadcastchannel/basics.any.js:
```diff
@@ -1,3 +1,11 @@
+      () => BroadcastChannel(""),
+      "Calling BroadcastChannel constructor without 'new' must throw"
+    );
+});
```
Add description

inexorabletash reviewed (commented):
Overall LGTM, just lots of nitpicks. Sorry for being a broken record about the test name, but it's important when reviewing the results when something does go wrong.

JakeChampion commented:
@inexorabletash Thanks for the thorough review - I believe I have addressed all the feedback now ☺️

inexorabletash reviewed (approved):
LGTM, thank you!!!
